File: mgsAnalysis/initialization.m
function [p, taskMap] = initialization(p, analysis_type, prac_status)

% Adding all the folders and the helper function
tmp = pwd; tmp2 = strfind(tmp,filesep);
p.master = tmp(1:(tmp2(end)-1));
addpath(genpath(p.master)); 

% Check the system running on: currently accepted: syndrome and my macbook
[ret, hostname] = system('hostname');
if ret ~= 0
    hostname = getenv('HOSTNAME');
end
p.hostname = strtrim(hostname);

% Initialize all the paths
if strcmp(p.hostname, 'syndrome') || strcmp(p.hostname, 'vader') || ...
        strcmp(p.hostname, 'zod') || strcmp(p.hostname, 'zod.psych.nyu.edu') || strcmp(p.hostname, 'loki.psych.nyu.edu')% If running on Syndrome or Vader or Zod
    p.datc = '/d/DATD/datd/MD_TMS_EEG';
    p.EEGData = [p.datc '/EEGData/sub' p.subjID '/day' num2str(p.day, '%02d')];
    p.fieldtrip = '/d/DATA/hyper/software/fieldtrip-20220104/';
else % If running on World's best MacBook
    p.datc = '/Users/mrugankdake/Documents/Clayspace/EEG_TMS/datd/MD_TMS_EEG';
    p.EEGData = [p.datc '/EEGData/sub' p.subjID '/day' num2str(p.day, '%02d')];
    p.fieldtrip = '/Users/mrugankdake/Documents/MATLAB/fieldtrip-20220104/';
end
p.data = [p.datc '/data'];
addpath(genpath(p.data));

p.phosphene = [p.data '/phosphene_data/sub' p.subjID];
if prac_status == 1
    p.mgs = [p.data '/mgs_practice_data/sub' p.subjID];
    p.analysis = [p.datc '/analysis/practice'];
else
    p.mgs = [p.data '/mgs_data/sub' p.subjID];
    p.analysis = [p.datc '/analysis'];
end
addpath(genpath(p.analysis));
p.dayfolder = [p.mgs '/day' num2str(p.day, '%02d')];

if prac_status == 1
    taskMapfilename = [p.phosphene '/taskMapPractice_sub' p.subjID '_antitype_mirror.mat'];
    load(taskMapfilename);
    taskMap = taskMapPractice;
    p.tmstatus = 0; %0 if off, 1 if on
else
    taskMapfilename = [p.phosphene '/taskMap_sub' p.subjID '_day' num2str(p.day, '%02d') '_antitype_mirror.mat'];
    load(taskMapfilename);
    p.tmstatus = taskMap(1).TMScond; %0 if off, 1 if on
end


% Folder to save analysis data
p.save = [p.analysis '/calib/sub' p.subjID '/day' num2str(p.day, '%02d')];
p.save_eyedata = [p.save '/EyeData'];
if ~exist(p.save, 'dir')
    mkdir(p.save);
end

% Add toolbox to path (either iEye or fieldtrip)
if strcmp(analysis_type, 'eye')
    p.iEye = [p.master '/iEye'];
    addpath(genpath(p.iEye));
elseif strcmp(analysis_type, 'eeg')
    p.saveEEG = [p.datc '/EEGfiles'];
    addpath(p.fieldtrip);
    ft_defaults;
end
end

File: mgsAnalysis/analysisEEG/initialization.py
import os, socket
from shutil import copyfile


def copyFiles(from_path, to_path):
    f_list = os.listdir(from_path)
    f_list = [f for f in f_list if f.endswith('.vhdr') or f.endswith('.vmrk') or f.endswith('.eeg')]
    for f in f_list:
        copyfile(os.path.join(from_path, f), os.path.join(to_path, f))
    

def load_paths(subjID, day):
    p = {}
    hostname = socket.gethostname()
    if hostname == 'syndrome' or hostname == 'zod.psych.nyu.edu' or hostname == 'zod':
        p['datd'] = '/d/DATD/datd/MD_TMS_EEG'
    elif hostname == 'vader':
        p['datd'] = '/clayspace/datd/MD_TMS_EEG'
    else:
        p['datd'] = '/Users/mrugankdake/Documents/Clayspace/EEG_TMS/datd/MD_TMS_EEG'
    
    p['data'] = os.path.join(p['datd'], 'data')
    p['analysis'] = os.path.join(p['datd'], 'analysis')
    p['meta'] = os.path.join(p['analysis'], 'meta_analysis')
    p['EEGData'] = f"{p['datd']}/EEGData/sub{subjID:02d}/day{day:02d}"
    p['EEGfiles'] = f"{p['datd']}/EEGfiles/sub{subjID:02d}/day{day:02d}"
    # if not os.path.exists(p['EEGfiles']):
    #     os.makedirs(p['EEGfiles'])
        # copyFiles(p['EEGData'], p['EEGfiles'])
    
    p['EEGpy'] = f"{p['datd']}/EEGpy/sub{subjID:02d}/day{day:02d}"

    if not os.path.exists(p['EEGpy']):
        os.makedirs(p['EEGpy'])
    p['EEGroot'] = f"{p['EEGpy']}/sub{subjID:02d}_day{day:02d}"
    p['fif_path'] = f"{p['EEGroot']}_raw.fif"
    # p['raw_fpath'] = f"{p['EEGroot']}/sub{subjID:02d}_day{day:02d}.vhdr"
    # if ~os.path.exists(p['raw_fpath']):

    return p
